sdk/typespec/typespec_client_core/src/http/headers/mod.rs

```diff
@@ -273,3 +325,85 @@ impl From<&String> for HeaderValue {
         s.clone().into()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use typespec::error::ErrorKind;
+    use url::Url;
```

Comment:
Nit: I recommend using our re-export in case we end up defining this ourselves, which I'm leaning toward to reduce dependencies. Same for `http_types` and some others as well.